openapi/resource-server.yaml

```diff
@@ -200,6 +201,10 @@ paths:
                 externalRef:
                   type: string
                   description: A reference that can be used by external systems to reconcile this payment with their systems. E.g. An invoice number. (Optional)
+                metadata:
+                  type: string
+                  format: json
+                  description: Additional metadata associated with the incoming payment, represented as a JSON string. (Optional)
```

Comment:
Is `format: json` valid? Not sure if we need this if `type: string`.

If we want to limit it to an object with free-form fields, we could also do this instead:
```yaml
    metadata:
      type: object
      additionalProperties: true
```

We could discuss in slack with Sabine as well

Reply:
I double-checked and json is not a valid format. I will remove if not going the object route.

In graphql we need to use a string (no "any" type input). That's mostly what informed my direction here but I suppose if we want to enforce the metadata as an object then the object type would be better.

I think it probably is a good idea that we ensure the metadata is a valid object. That was the original use-case and I think it would just make more sense when we get incoming payments (no logic to determine what metadata is).

Reply:
And on the topic of graphql, I guess an arbitrary object is possible with something like [graphql-type-json](https://www.npmjs.com/package/graphql-type-json)

Reply:
switching to an object here and will implement it as a json scalar in gql using `graphql-scalars` (a package we already use)